clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -157,4 +290,42 @@ public KafkaBasedCommandSender(
 
         return props;
     }
+
+    private Future<Void> subscribeForCommandResponse(final String tenantId,
+            final Handler<DownstreamMessage<KafkaMessageContext>> messageHandler, final Span span) {
+        if (commandResponseSubscriptions.get(tenantId) != null) {
+            span.log("command response consumer already exists");
+            LOGGER.debug("command response consumer already exists for tenant[{}]", tenantId);
+            return Future.succeededFuture();
+        } else {
+            final Map<String, String> consumerConfig = this.consumerConfig
+                    .getConsumerConfig(HonoTopic.Type.COMMAND_RESPONSE.toString());
+
+            // Use a unique group-id so that all command responses for this tenant are received by this consumer.
+            // Thereby the responses can be correlated with the command that has been sent.
+            consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, tenantId + "-" + UUID.randomUUID().toString());
+
+            return Optional.ofNullable(commandResponseConsumerFactory)
+                    .map(t -> t.apply(messageHandler))
+                    .orElseGet(
+                            () -> KafkaBasedDownstreamMessageConsumer.create(
```

Comment:
If I understood correctly this call also starts the consumer. Maybe I missed it, but I couldn't find a call to `KafkaBasedDownstreamMessageConsumer`'s `close()` method.

Is that still missing or does the consumer stop in some other way?

Reply:
The consumers are closed inside the `stop()` method.

Reply:
But that `stop()` method isn't called when the command response was processed successfully or failed? At least I could not find a reference in `KafkaBasedCommandSender` to it.

Reply:
The consumers are created for every _new tenant_ and I think its a disadvantage to close and create a new consumer for every command being sent. So they are not closed after every command.

Reply:
Ah! I see, that's done in line 296. Thanks! 😄